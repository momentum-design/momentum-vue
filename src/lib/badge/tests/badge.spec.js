import { shallowMount } from '@vue/test-utils';
import Badge from '../index.vue';

describe('Badge', () => {

  it('should match snapshot', () => {
    const wrapper = shallowMount(Badge);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should render one Child Div', () => {
    const wrapper = shallowMount(Badge, {
      slots: { default: '<div>Test</div>' }
    });
    expect(wrapper.find('div').contains('div')).toBe(true);
    expect(wrapper.find('div').text()).toBe('Test');
  });

  it('should assign conditional rounded class name', () => {
    const wrapper = shallowMount(Badge, {
      propsData: { rounded: true }
    });
    expect(wrapper.find('div').classes('md-badge--round')).toBe(true);
  });

  it('should not assign conditional rounded class name', () => {
    const wrapper = shallowMount(Badge);
    expect(wrapper.find('div').classes('md-badge--round')).toBe(false);
  });

  it('should assign conditional color class name', () => {
    const wrapper = shallowMount(Badge, {
      propsData: { color: 'blue' }
    });
    expect(wrapper.find('div').classes('md-badge--blue')).toBe(true);
  });

  it('should not assign conditional color class name', () => {
    const wrapper = shallowMount(Badge);
    expect(wrapper.find('div').classes('md-badge--blue')).toBe(false);
  });

});
