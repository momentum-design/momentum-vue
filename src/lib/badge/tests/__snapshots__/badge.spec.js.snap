// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge should match snapshot 1`] = `<div class="md-badge md-badge--undefined"></div>`;
